usr/share/rear/rescue/default/500_ssh.sh

```diff
@@ -3,164 +3,97 @@
-                                    LogUserOutput ""
-                                    user_input="$( UserInput -I SSH_PRIVATE_KEYS_UNPROTECTED_CONFIRMATION \
-                                        -t 0 -p "Type 'yes' if you want to allow unauthenticated access" )"
-                                    if [ "$user_input" == "yes" ]; then
-                                        LogUserOutput "SSH private keys will be copied in unprotected form"
-                                        break
-                                    fi
-                                    ;;
-
-                                *)
-                                    LogUserOutput "Repeat SSH client passphrase: \c"
-                                    read -s repeated_passphrase <&6
-                                    UserOutput ""
-                                    if [ "$passphrase" != "$repeated_passphrase" ]; then
-                                        LogUserOutput "Passphrases do not match"
-                                        passphrase=""
-                                    else
-                                        LogUserOutput "SSH private keys will be copied and passphrase-protected"
-                                        break
-                                    fi
-                                    ;;
-                            esac
-                        done
-                    fi
-
-                    # Carry out the user's decision
-                    case "$passphrase" in
-                        "")
-                            Log "Ignoring unprotected private key $original_file"
-                            rm "$target_file"
-                            ;;
-
-                        UnProtected)
-                            Log "Copying unprotected private key $original_file"
-                            ;;
-
-                        *)
-                            Log "Copying private key $original_file, adding passphrase-protection"
-                            if ! ssh-keygen -p -P '' -N "$passphrase" -f "$target_file"; then
-                                LogPrintError "Could not assign the passphrase to $target_file"
-                                Error "Stopping to avoid leaking unprotected SSH private keys onto the rescue medium"
-                            fi
-                    esac
+local target_dir="$ROOTFS_DIR/root/.ssh" original_file target_file
+
+mkdir -p -m u=rwx,go=- "$target_dir"  # COPY_AS_IS will take effect later, so create the target .ssh directory first
+
+for original_file in /root/.ssh/id_* /root/.ssh/known_host[s]; do
+    target_file="$target_dir/$(basename "$original_file")"
+    cp -p "$original_file" "$target_file"
+
+    case "$original_file" in
+        (*/id_*.pub|*/known_hosts)
+            # Public key files or 'knownhosts' contain no secrets and can be copied as-is
+            Log "Copying $original_file"
```

Comment:
I wouldn't log this. It is only interesting for you right now while you develop it. I would log the encryption and the deletion because it is something beyond just copying. In verbose mode the copied files show up in the log in any case.